src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -0,0 +1,273 @@
+      // and then filter on the resulting boolean.
+      jit_operator->add_jit_operator(std::make_shared<JitFilter>(expression->result()));
+    }
+
+    // Identify the top-most projection node to determine the output columns
+    auto top_projection = node;
+    while (top_projection != input_node && top_projection->type() != LQPNodeType::Projection) {
+      top_projection = top_projection->left_input();
+    }
+
+    // Add a compute operator for each output column that computes the column value.
+    auto write_table = std::make_shared<JitWriteTuple>();
+    for (const auto& output_column :
+         boost::combine(top_projection->output_column_names(), top_projection->output_column_references())) {
+      const auto expression = _translate_to_jit_expression(output_column.get<1>(), *read_tuple, input_node);
+      // It the JitExpression is of type ExpressionType::Column, there is no need to add a compute node, since it
+      // would not compute anything anyway
+      if (expression->expression_type() != ExpressionType::Column) {
+        jit_operator->add_jit_operator(std::make_shared<JitCompute>(expression));
+      }
+      write_table->add_output_column(output_column.get<0>(), expression->result());
+    }
+    jit_operator->add_jit_operator(write_table);
+    return jit_operator;
+  } catch (const std::exception& exception) {
+    // If JitOperatorWrapper creation fails, we fall back to the regular LQPTranslator
+    return LQPTranslator::translate_node(node);
+  }
+}
+
+std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_translate_to_jit_expression(
+    const std::shared_ptr<AbstractLQPNode>& node, JitReadTuple& jit_source,
+    const std::shared_ptr<AbstractLQPNode>& input_node) const {
+  std::shared_ptr<const JitExpression> left, right;
+  switch (node->type()) {
+    case LQPNodeType::Predicate:
+      // If there are further conditions (either PredicateNodes of UnionNodes) down the line, we need a logical AND here
+      if (_has_another_condition(node->left_input())) {
+        left = _translate_to_jit_expression(std::dynamic_pointer_cast<PredicateNode>(node), jit_source, input_node);
+        right = _translate_to_jit_expression(node->left_input(), jit_source, input_node);
+        return std::make_shared<JitExpression>(left, ExpressionType::And, right, jit_source.add_temporary_value());
+      } else {
+        return _translate_to_jit_expression(std::dynamic_pointer_cast<PredicateNode>(node), jit_source, input_node);
+      }
+
+    case LQPNodeType::Union:
+      left = _translate_to_jit_expression(node->left_input(), jit_source, input_node);
+      right = _translate_to_jit_expression(node->right_input(), jit_source, input_node);
+      return std::make_shared<JitExpression>(left, ExpressionType::Or, right, jit_source.add_temporary_value());
+
+    case LQPNodeType::Projection:
+      // We don't care about projection nodes here, since they do not perform any tuple filtering
```

Comment:
Is it really safe to discard Projections? They could compute values used higher up in the LQP, after all.

Reply:
It is safe in this place, since we are only interested in tuple filtering here.
If we work on a computed column value, this will be handled when translating the corresponding `LQPColumnReference`.